### PR TITLE
mfem+dependencies update

### PR DESCRIPTION
**Pull request by v-dobrev**
Update the `mfem` package with more variants while extending some of the packages it (optionally) depends on.

In addition, the properties `headers` and `libs` are defined for a number of related packages and the proposed new properties `all_headers` and `all_libs` (in PR #7256) are defined and used.

CC: (package maintainers of affected packages)
- mfem: @goxberry, @tzanio, @markcmiller86, @acfisher
- hypre: @tgamblin ?
- sundials: @cswoodward, @gardner48
- petsc: @balay, @BarrySmith
- mumps: ?
- netlib-scalapack: ?
- suite-sparse: ?
- laghos: @bhatele ?
- superlu-dist: @goxberry ?
- xsdk: @balay, @BarrySmith
- gnutls, libunwind, openblas, openmpi, zlib: ?

### CODE REVIEW

File: var/spack/repos/builtin/packages/mumps/package.py [hunk 1]
```diff
@@ -262,3 +262,11 @@ def install(self, spec, prefix):
                 os.system('./dsimpletest < input_simpletest_real')
                 if '+complex' in spec:
                     os.system('./zsimpletest < input_simpletest_cmplx')
+
+    @property
+    def libs(self):
```
Comment (davydden) [inline var/spack/repos/builtin/packages/mumps/package.py]: Can u please split this PR in small independent ones? Openblas, openmpi, hypre, etc — those changes are independent

**Comment by goxberry** [comment]
Agreed; this convention is mentioned in the [Contribution Guide](https://spack.readthedocs.io/en/latest/contribution_guide.html).

**Comment by v-dobrev** [comment]
OK, I will split this PR into separate PRs for each package.

I was under the impression (judging by the last xSDK release) that it is OK to have PRs that modify multiple packages. I guess that was not done exactly by the book. 😃

I'll close this PR once I have all the small PRs created.

**Comment by davydden** [comment]
> I was under the impression (judging by the last xSDK release) that it is OK to have PRs that modify multiple packages.

it kind of depends, in your case you certainly can split it as many things are independent. It makes reviewing PRs much much more easier.

**Comment by davydden** [comment]
since you already split this PR into multiple, i think we can safely close it and look into those separate ones.

**Pull request closed without merging**